File: PYTHON/Baju.py
from Aksesoris import Aksesoris  # Mengimpor kelas induk Aksesoris

class Baju(Aksesoris):  # Membuat subclass Baju yang mewarisi Aksesoris
    
    def __init__(self, id="", nama_produk="", harga_produk=0, stok_produk=0, jenis="", bahan="", warna="", untuk="", size="", merk=""):
        super().__init__(id, nama_produk, harga_produk, stok_produk, jenis, bahan, warna)  # Memanggil konstruktor kelas induk
        # Konstruktor untuk menginisialisasi atribut dengan nilai default
        self.set_untuk(untuk) 
        self.set_size(size)  
        self.set_merk(merk)  

    def set_untuk(self, untuk):
        self._untuk = untuk  # Menyimpan kategori pengguna baju
    
    def get_untuk(self):
        return self._untuk  # Mengembalikan kategori pengguna baju

    def set_size(self, size):
        self._size = size  # Menyimpan ukuran baju
    
    def get_size(self):
        return self._size  # Mengembalikan ukuran baju
    
    def set_merk(self, merk):
        self._merk = merk  # Menyimpan merk baju
    
    def get_merk(self):
        return self._merk  # Mengembalikan merk baju
    
    def display(self, untuk_max, size_max, merk_max):
        # Menampilkan data produk dengan format yang rapi berdasarkan panjang maksimal kolom
        print(
            self._untuk + " " * (untuk_max - len(self._untuk)),
            self._size + " " * (size_max - len(self._size)),
            self._merk + " " * (merk_max - len(self._merk)),
        )

File: JAVA/Baju.java
// Kelas Baju merupakan subclass dari Aksesoris
public class Baju extends Aksesoris {
    // Deklarasi atribut tambahan khusus untuk baju
    private String untuk; // Jenis hewan yang menggunakan baju (misal: anjing, kucing)
    private String size;  // Ukuran baju (S, M, L, XL)
    private String merk;  // Merk baju

    // Constructor default untuk menginisialisasi atribut dengan nilai awal kosong
    public Baju() {
        super(); // Memanggil constructor default dari superclass (Aksesoris)
        this.untuk = "";
        this.size = "";
        this.merk = "";
    }

    // Constructor dengan parameter untuk mengisi semua atribut saat objek dibuat
    public Baju(String id, String namaProduk, int hargaProduk, int stokProduk, 
                String jenis, String bahan, String warna, 
                String untuk, String size, String merk) {
        super(id, namaProduk, hargaProduk, stokProduk, jenis, bahan, warna); // Memanggil constructor superclass
        this.untuk = untuk; // Mengisi atribut 'untuk' sesuai parameter
        this.size = size;   // Mengisi atribut 'size' sesuai parameter
        this.merk = merk;   // Mengisi atribut 'merk' sesuai parameter
    }

    // Getter untuk mendapatkan nilai atribut 'untuk'
    public String getUntuk() {
        return untuk;
    }

    // Setter untuk mengubah nilai atribut 'untuk'
    public void setUntuk(String untuk) {
        this.untuk = untuk;
    }

    // Getter untuk mendapatkan nilai atribut 'size'
    public String getSize() {
        return size;
    }

    // Setter untuk mengubah nilai atribut 'size'
    public void setSize(String size) {
        this.size = size;
    }

    // Getter untuk mendapatkan nilai atribut 'merk'
    public String getMerk() {
        return merk;
    }

    // Setter untuk mengubah nilai atribut 'merk'
    public void setMerk(String merk) {
        this.merk = merk;
    }

    // Overriding method display untuk menampilkan informasi produk baju
    public void display(int idMax, int namaMax, int hargaMax, int stokMax, 
                        int jenisMax, int bahanMax, int warnaMax, 
                        int untukMax, int sizeMax, int merkMax) {
        super.display(idMax, namaMax, hargaMax, stokMax, jenisMax, bahanMax, warnaMax); // Memanggil method display dari superclass
        System.out.printf("%-" + untukMax + "s %-" + sizeMax + "s %-" + merkMax + "s\n", untuk, size, merk); // Menampilkan atribut tambahan khusus baju
    }
}
